### DISCUSSION TITLE
Mapping parameters

### DISCUSSION THREAD

**Discussion by kevin5km** [scottoffen/grapevine#82] [Q&A] [answered]
Are there any examples of mapping path parameters? I've done this but it didn't work. Are there any other things I have to set up? 

```csharp
[RestRoute("Get", "/api/heartbeat/{requestID}")]
public async Task Heartbeat(IHttpContext context)
{
    var requestID = context.Request.PathParameters["requestID"];
    await context.Response.SendResponseAsync(requestID).ConfigureAwait(false);
}
```

I'm using the path and query string `/api/heartbeat?requestID=ab123`.

**Comment by scottoffen** [comment, marked as answer]
Path parameters are different from the query string.

If you want the request id in the path parameter, your path should be `/api/heartbeat/ab123`, and you would use the your existing method, with a modification to the `Route` attribute:

```csharp
[RestRoute("Get", "/api/heartbeat/{requestID}")]
public async Task Heartbeat(IHttpContext context)
{
    var requestID = context.Request.PathParameters["requestID"];
    await context.Response.SendResponseAsync(requestID).ConfigureAwait(false);
}
```

If you want the request id to be in the query string, you would use your existing path and query string of `/api/heartbeat?requestID=ab123`, but your `Route` attribute and method would change to:

```csharp
[RestRoute("Get", "/api/heartbeat")]
public async Task Heartbeat(IHttpContext context)
{
    var requestID = context.Request.QueryString["requestID"];
    await context.Response.SendResponseAsync(requestID).ConfigureAwait(false);
}
```

You don't want to include the query string in the path pattern.
